lib/maths/CBoostedTreeImpl.cc

```diff
@@ -878,32 +891,43 @@ CBoostedTreeImpl::trainTree(core::CDataFrame& frame,
         featureSampleProbabilities = m_FeatureSampleProbabilities;
         this->featureBag(featureSampleProbabilities, featureBag);
 
+        std::size_t n{splitCandidateTreeNodes.size()};
+        std::size_t currentNumberInternalNodes{(tree.size() - 1) / 2};
+        std::size_t lastCandidateIdx{n - (maximumNumberInternalNodes - currentNumberInternalNodes)};
```

Comment:
Doesn't this underflow in a bad way? I think this needs to be:
```suggestion
        auto smallestCurrentCandidateGainIndex = static_cast<std::ptrdiff_t>(numberSplittableLeaves) - static_cast<std::ptrdiff_t>(maximumNumberInternalNodes - currentNumberInternalNodes);
```

Reply:
You are right, it was `int` in the original version, I changed it to `size_t` for clean-up, but it should be `ptrdiff_t` of course.